contribs/gmf/src/directives/objectediting.js

```diff
@@ -270,80 +268,58 @@ gmf.ObjecteditingController = function($scope, $timeout, gettextCatalog,
   this.process = gmf.ObjecteditingtoolsController.ProcessType.ADD;
 
   /**
-   * @type {null|ol.layer.Image|ol.layer.Tile}
+   * @type {?ol.layer.Image|ol.layer.Tile}
    * @private
    */
   this.editableWMSLayer_ = null;
 
   /**
-   * @type {jsts.io.OL3Parser}
+   * @type {!jsts.io.OL3Parser}
    * @private
    */
   this.jstsOL3Parser_ = new jsts.io.OL3Parser();
 
-  const geometry = this.feature.getGeometry();
-
   /**
    * The state of the feature determines whether the next 'save' request
    * should be an 'insert' or 'update' one.
-   * @type {string}
+   * @type {string|undefined}
    * @private
    */
-  this.state_ = geometry ? gmf.ObjecteditingController.State.UPDATE :
-    gmf.ObjecteditingController.State.INSERT;
+  this.state_;
 
   /**
-   * @type {Array.<?ol.geom.Geometry>}
+   * @type {!Array.<!ol.geom.Geometry>}
    * @private
    */
   this.geometryChanges_ = [];
```

Comment:
This array is supposed to be one or null and geometry types.  This change broke the tool.